src/google/protobuf/compiler/csharp/csharp_source_generator_base.cc

```diff
@@ -56,6 +56,10 @@ SourceGeneratorBase::~SourceGeneratorBase() {
 }
 
 void SourceGeneratorBase::WriteGeneratedCodeAttributes(io::Printer* printer) {
+  printer->Print("[global::System.Diagnostics.DebuggerNonUserCodeAttribute]\n");
+}
+
+void SourceGeneratorBase::WriteGeneratedTypeAttributes(io::Printer* printer) {
```

Comment:
My reasoning for keeping it was that we've already done the work to make sure it's called at all the right places. But I don't mind removing it - let's just make sure we keep that removal as a separate commit which is easy to revert. Will add that commit to this PR.